src/main/java/io/cryostat/net/web/http/api/beta/graph/ActiveRecordingsFetcher.java

```diff
@@ -0,0 +1,133 @@
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import io.cryostat.net.web.http.api.beta.graph.RecordingsFetcher.Recordings;
+
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+
+class ActiveRecordingsFetcher implements DataFetcher<List<GraphRecordingDescriptor>> {
+
+    @Inject
+    ActiveRecordingsFetcher() {}
+
+    public List<GraphRecordingDescriptor> get(DataFetchingEnvironment environment)
+            throws Exception {
+        Recordings source = environment.getSource();
+        List<GraphRecordingDescriptor> result = new ArrayList<>(source.active);
+
+        FilterInput filter = FilterInput.from(environment);
+        if (filter.contains(FilterInput.Key.NAME)) {
+            String recordingName = filter.get(FilterInput.Key.NAME);
+            result =
+                    result.stream()
+                            .filter(r -> Objects.equals(r.getName(), recordingName))
+                            .collect(Collectors.toList());
+        }
+        if (filter.contains(FilterInput.Key.STATE)) {
+            String state = filter.get(FilterInput.Key.STATE);
+            result =
+                    result.stream()
+                            .filter(r -> Objects.equals(r.getState().name(), state))
+                            .collect(Collectors.toList());
+        }
+        if (filter.contains(FilterInput.Key.CONTINUOUS)) {
+            boolean continuous = filter.get(FilterInput.Key.CONTINUOUS);
+            result =
+                    result.stream()
+                            .filter(r -> Objects.equals(r.isContinuous(), continuous))
+                            .collect(Collectors.toList());
+        }
+        if (filter.contains(FilterInput.Key.TO_DISK)) {
+            boolean toDisk = filter.get(FilterInput.Key.TO_DISK);
+            result =
+                    result.stream()
+                            .filter(r -> Objects.equals(r.getToDisk(), toDisk))
+                            .collect(Collectors.toList());
+        }
+        if (filter.contains(FilterInput.Key.DURATION_GE)) {
+            long duration = filter.get(FilterInput.Key.DURATION_GE);
+            result =
+                    result.stream()
+                            .filter(r -> r.getDuration() >= duration)
+                            .collect(Collectors.toList());
+        }
+        if (filter.contains(FilterInput.Key.DURATION_LE)) {
+            long duration = filter.get(FilterInput.Key.DURATION_LE);
+            result =
+                    result.stream()
+                            .filter(r -> r.getDuration() <= duration)
+                            .collect(Collectors.toList());
+        }
+        if (filter.contains(FilterInput.Key.START_TIME_BEFORE)) {
+            long startTime = filter.get(FilterInput.Key.START_TIME_BEFORE);
+            result =
+                    result.stream()
+                            .filter(r -> r.getStartTime() <= startTime)
```

Comment:
The filter `startTimeBeforeEqual` matches if the `startTime` value given is before or equal to a recording's start time, right? Shouldn't the equality be `r.getStartTime() >= startTime` then?

Reply:
It's meant to be "if the recording was started at or before the given timestamp", but I guess the filter name isn't really clear about that. Any suggestion for a better name?

Reply:
Nevermind, I realize it matches if a recording's start time is before or equal to the filter `startTime` value.

Reply:
I think it's fine as is actually. More of a misunderstanding on my part than any confusion in the filter name.